core/src/datamatrix/DMDecoder.cpp

```diff
@@ -573,7 +573,7 @@ CorrectErrors(ByteArray& codewordBytes, int numDataCodewords)
 
 	// Copy back into array of bytes -- only need to worry about the bytes that were data
 	// We don't care about errors in the error-correction codewords
-	std::copy_n(codewordsInts.begin(), numDataCodewords, codewordBytes.begin());
+	std::transform(codewordsInts.begin(), codewordsInts.begin() + numDataCodewords, codewordBytes.begin(), [](int const& cw) { return static_cast<uint8_t>(cw); });
 
```

Comment:
E.g. here the comment explicitly states that we want to copy ints into bytes and the transform with the cast is obviously less readable (see comment above about the casting).

Reply:
But in reality we are transforming `int`s to fit in an 8bit location. Again, just my point of view.

Reply:
I agree with your assessment. But I (personally) don't see the real benefit. If we are sure the implicit narrowing is without issues (there is no value outside of 0..255) then this only reduces readability. If we are not sure about that, then this does nothing more than the implicit narrowing except it suggest, that we actually want it to mask out higher bits (which we don't). In that second case, we should actually check that all values are valid. One would have to inspect the code and see if that narrowing can or can not fail (then the check would be unnecessary and misleading as well).

Reply:
OK, withdrawing on this.

Reply:
Added eb9a47632cdf3ee0a317e2992332f59947f16094.